classes/Group.php

```diff
@@ -218,11 +218,28 @@ public function delete()
             return false;
         }
         if (parent::delete()) {
+            $cartRuleIds = Db::getInstance()->executeS('SELECT `id_cart_rule`
+                FROM `' . _DB_PREFIX_ . 'cart_rule_group`
+                WHERE `id_group` = ' . (int) $this->id);
+
             Db::getInstance()->execute('DELETE FROM `' . _DB_PREFIX_ . 'cart_rule_group` WHERE `id_group` = ' . (int) $this->id);
             Db::getInstance()->execute('DELETE FROM `' . _DB_PREFIX_ . 'customer_group` WHERE `id_group` = ' . (int) $this->id);
             Db::getInstance()->execute('DELETE FROM `' . _DB_PREFIX_ . 'category_group` WHERE `id_group` = ' . (int) $this->id);
             Db::getInstance()->execute('DELETE FROM `' . _DB_PREFIX_ . 'group_reduction` WHERE `id_group` = ' . (int) $this->id);
             Db::getInstance()->execute('DELETE FROM `' . _DB_PREFIX_ . 'product_group_reduction_cache` WHERE `id_group` = ' . (int) $this->id);
+
+            if (is_array($cartRuleIds) && count($cartRuleIds)) {
+                $cartRuleIds = array_unique(array_map('intval', array_column($cartRuleIds, 'id_cart_rule')));
+                foreach ($cartRuleIds as $cartRuleId) {
+                    $cartRule = new CartRule($cartRuleId);
+                    if (Validate::isLoadedObject($cartRule)) {
+                        $cartRule->delete();
+                    } else {
+                        Db::getInstance()->delete('cart_rule', 'id_cart_rule = ' . (int) $cartRuleId);
+                    }
+                }
+            }
```

Comment:
It should turn off affected cart rules instead, ideally with an additional description added to the internal description of the cart rule.

Reply:
I still think this should not be placed in the legacy code, and especially not be applied all the time It should be conditionned by the feature flag, moving this logic in the CQRS handler will naturally bind it to the feature flag